simd/cpu_amd64.s

```diff
@@ -0,0 +1,55 @@
+/* This file is a copy of https://cs.opensource.google/go/x/sys/+/master:cpu/cpu_x86.s.
+Here is its license, which only applies to the copied parts and not to the rest of chromem-go,
+which is licensed under the GNU Affero General Public License.
+
+BSD 3-clause license
+
+Copyright (c) 2009 The Go Authors. All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are
+met:
+
+   * Redistributions of source code must retain the above copyright
+notice, this list of conditions and the following disclaimer.
+   * Redistributions in binary form must reproduce the above
+copyright notice, this list of conditions and the following disclaimer
+in the documentation and/or other materials provided with the
+distribution.
+   * Neither the name of Google Inc. nor the names of its
+contributors may be used to endorse or promote products derived from
+this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+#include "textflag.h"
```

Comment:
Do you know if this could lead to any constraints on users side? Does it work when using [TinyGo](https://github.com/tinygo-org/tinygo) as compiler? (Haven't tried with current `chromem-go` either though)

Reply:
> Do you know if this could lead to any constraints on users' side?

not really, as long as we take care of proper build tags.

> Does it work when using [TinyGo](https://github.com/tinygo-org/tinygo) as compiler?

iirc,
- TinyGo doesn't currently enable support for AVX instructions,
- TinyGo does not support Go's Assembler, so we should skip building those files

If we want to attempt to build TinyGo "normal" binaries we can manage the builds with the `tinygo` build tag so building on amd64 with TinyGo will just skip the custom code. 
I will take care of that.

However I don't think tinygo works right now (unrelated to the changes being worked on here). Tried quickly to build the _minimal_ example with TinyGo and it panics
```sh
checkId: 22 should be 23
panic: bootstrap type wrong id: mapType mapType not <nil>
```

Reply:
It would be `// +build !tinygo` in both this assembly file and the `cpu_amd64.go` one, right?